### PR TITLE
Add pki ticket exchange / Envendpoint

### PR DESCRIPTION
**Pull request by tsupertramp**

### CODE REVIEW

**Comment by vkhatri** [comment]
@tsupertramp Thank you for the PR! LGTM

File: libraries/provider_environment.rb [hunk 1]
```diff
@@ -41,12 +41,8 @@ def whyrun_supported?
...
         search_pattern = new_resource.search_pattern || "chef_environment:#{new_resource.environment}"
         if new_resource.limit_region && !new_resource.server_region
-          if node.key?('ec2')
-            server_region = node['ec2']['placement_availability_zone'].chop
-          else
-            # add more cloud providers
-            server_region = nil
-          end
+          server_region = nil
+          server_region node['ec2']['placement_availability_zone'].chop if node.key?('ec2')
```
Comment (vkhatri) [inline libraries/provider_environment.rb]: `new_resource.server_region` is ignored, its a pre existing bug in the condition and server_region is missing assignment operator. i will update the if block to -

```ruby
         search_pattern = new_resource.search_pattern || "chef_environment:#{new_resource.environment}"
-        if new_resource.limit_region && !new_resource.server_region
-          server_region = nil
-          server_region node['ec2']['placement_availability_zone'].chop if node.key?('ec2')
+        server_region = new_resource.server_region
+
+        if new_resource.limit_region && !server_region
+          server_region = node['ec2']['placement_availability_zone'].chop if node.key?('ec2')
         end
```

File: attributes/default.rb [hunk 2]
```diff
@@ -71,18 +67,16 @@
...
   default['icinga2']['service_config_file'] = '/etc/sysconfig/icinga2'
 
-  case node['kernel']['machine']
-  when 'x86_64'
-    default['icinga2']['plugins_dir'] = '/usr/lib64/nagios/plugins'
-  else
-    default['icinga2']['plugins_dir'] = '/usr/lib/nagios/plugins'
-  end
+  default['icinga2']['plugins_dir'] = if node['kernel']['machine'] == 'x86_64'
+                                        '/usr/lib/nagios/plugins'
```
Comment (vkhatri) [inline attributes/default.rb]: `x86_64` == `/usr/lib64/nagios/plugins`

**Comment by vkhatri** [comment]
Found some typos, will fix it before merging to upstream. Giving it a quick run.

**Comment by vkhatri** [comment]
@tsupertramp By default `create_endpoints, create_zones and create_pki_tickets` creates resources which might not be useful for environments configured with `nrpe` on clients. This should be optional and configurable by a node attribute `default['icinga2']['enable_env_pki'] = false`. Will add this attribute.

**Comment by tsupertramp** [comment]
Thank your for fixing bugs, creating features and merging 👍

**Pull request merged**